crates/sui-bridge-indexer/src/indexer_builder.rs

```diff
@@ -118,15 +134,25 @@ impl<P, D, M> Indexer<P, D, M> {
         // Start latest checkpoint worker
         // Tasks are ordered in checkpoint descending order, realtime update task always come first
         // tasks won't be empty here, ok to unwrap.
-        let (live_task, backfill_tasks) = updated_tasks.split_first().unwrap();
-
-        let live_task_future = self.datasource.start_ingestion_task(
-            live_task.task_name.clone(),
-            live_task.checkpoint,
-            live_task.target_checkpoint,
-            self.storage.clone(),
-            self.data_mapper.clone(),
-        );
+        let backfill_tasks;
+        if self.disable_live_task {
+            backfill_tasks = updated_tasks;
+        } else {
+            let (_live_task, _backfill_tasks) = updated_tasks.split_first().unwrap();
+
+            backfill_tasks = _backfill_tasks.to_vec();
+            let live_task = _live_task;
+
+            let live_task_future = self.datasource.start_ingestion_task(
+                live_task.task_name.clone(),
+                live_task.checkpoint,
+                live_task.target_checkpoint,
+                self.storage.clone(),
+                self.data_mapper.clone(),
+            );
+
+            live_task_future.await?;
```

Comment:
if we await live task here, the back fill task will be blocked

Reply:
good call